crates/sdk/src/keygen/dummy.rs

```diff
@@ -55,7 +55,7 @@ pub(super) fn compute_root_proof_heights(
         public_values: vec![F::ZERO; num_user_public_values],
     };
     let vm = SingleSegmentVmExecutor::new(root_vm_config);
-    let res = vm.execute(root_exe, root_input.write()).unwrap();
+    let res = vm.execute(root_exe, root_input.write(), true).unwrap();
```

Comment:
is this the only place the `compute_heights` bool is used? I'm not sure where else the heights are used in this single segment result; if we actually don't use it anywhere else, I'd slightly prefer if we removed the bool arg and just handled it here

Reply:
We also call it in `execute_for_air_heights`. The rest of the calls (where we pass `false`) are in tests, so maybe we can just always compute heights in `SingleSegmentVmExecutor::execute`?

Reply:
We can't (easily) handle it here because `res` is of type `SingleSegmentVmExecutionResult`, so we don't have the `ExecutionSegment` to call `finalize_memory` on.

Reply:
I will rename `execute` to `execute_and_compute_heights` and always finalize.